app/controllers/api/v1x0/icons_controller.rb

```diff
@@ -23,30 +23,36 @@ def update
       end
 
       def raw_icon
-        image = find_icon(params.permit(:icon_id, :portfolio_item_id)).image.decoded_image
+        image = find_icon(params.permit(:icon_id, :portfolio_item_id, :portfolio_id)).image.decoded_image
         send_data(image,
                   :type        => MimeMagic.by_magic(image).type,
                   :disposition => 'inline')
       end
 
       def override_icon
-        overriden_icon = Catalog::OverrideIcon.new(params.require(:icon_id), params.require(:portfolio_item_id))
+        overriden_icon = Catalog::OverrideIcon.new(params.require(:icon_id), params.permit(:portfolio_id, :portfolio_item_id))
         render :json => overriden_icon.process.icon
       end
 
       private
 
       def icon_params
-        params.require([:content, :portfolio_item_id])
+        params.require(:content)
         icon_patch_params
       end
 
       def icon_patch_params
-        params.permit(:content, :source_ref, :source_id, :portfolio_item_id, :id)
+        params.permit(:content, :source_ref, :source_id, :portfolio_item_id, :portfolio_id)
```

Comment:
@lindgrenj6 Instead of using icon_patch_params you should be able to use this function from the common gem
https://github.com/ManageIQ/manageiq-api-common/blob/3848da0e2e6ec1395326d182cc97491620d0f668/lib/manageiq/api/common/application_controller_mixins/parameters.rb#L101

This will check the parameters in the openapi.json and only allow those parameters.

Before you use this function we need to add the mixin here
https://github.com/ManageIQ/catalog-api/blob/353bcf39383bc92987534f421a6c5054dda23a2a/app/controllers/application_controller.rb#L8
```
include ManageIQ::API::Common::ApplicationControllerMixins::Parameters
```

Reply:
I am up for doing this - but I would like to do it in a separate PR. That way I can take care of using `params_for_update` across every controller at once.